joomla/joomla-cms
# Create possibility for nested subforms

Pull Request for Issue #11551

### Summary of Changes
This PR adds the possibility to nest subforms into subforms, adding the possibility to stack them as many times as a user would like to.


### Testing Instructions
Create a form with a subform inside of a subform and test whether it works as expected (with multiple=true or false, layout=repeatable or repeatable-table, different min/max values, etc...).

A simple testcase is to testwise edit e.g. `plugins/fields/checkboxes/params/checkboxes.xml`, adding e.g. the following after [line 27](https://github.com/joomla/joomla-cms/blob/2f181e2ec0cb1a4c81000896367ea3bf90e7f878/plugins/fields/checkboxes/params/checkboxes.xml#L27):
```xml
<field
	name="subformlevel2"
	type="subform"
	label="PLG_FIELDS_CHECKBOXES_PARAMS_OPTIONS_LABEL"
	description="PLG_FIELDS_CHECKBOXES_PARAMS_OPTIONS_DESC"
	layout="joomla.form.field.subform.repeatable-table"
	icon="list"
	multiple="true"
>
	<form hidden="true" name="list_templates_modal" repeat="true">
		<field
			name="name_lvl2"
			type="text"
			label="PLG_FIELDS_CHECKBOXES_PARAMS_OPTIONS_NAME_LABEL"
			size="30"
		/>
	</form>
</field>
```

After that has been added, navigate to Content -> Fields -> New in the Administrator Backend, select "Checkboxes" as Type, and play around with the nested subforms.

To do some additional tests, play around with the XML definition (adding 3rd level subforms, changing the multiple/non-multiple stuff, required status, etc...) and take a look whether everything works as expected.

### Expected result
Something like in the following screenshot, where I have stacked 3 subforms inside of each other with just some basic textfields:
![image](https://user-images.githubusercontent.com/2082625/29329297-bc5dbb60-81f5-11e7-88ee-314e40b684c0.png)

### Additional information
This is a backwards-compatible change and hence does not require anything to be worried about (imho) - please correct me if I am wrong, maybe we can find a solution then.

I am aware that the main merge window of Joomla! 3.8 already has passed, so I guess this PR probably will target version 3.9 or 4.0, but I open this PR now because it is my first slightly bigger PR for Joomla, and I expect some feedback-cycles before we reach a merge-ready state.

I already have a question to begin with: I only committed the uncompressed JS file, is it my job to do the minification, or will some deploy-script take care of that?

Thanks in advance!
